### PR TITLE
Upgrade dependencies, add ruff

### PR DESCRIPTION
**Pull request by ojii**
Upgraded various dependencies
Added ruff checks
Fix functional test servers triggering macos firewall

### CODE REVIEW

**Comment by y26805** [comment]
should we fix CI first..? 👀 

> E: Failed to fetch http://deb.debian.org/debian/pool/main/g/golang-1.15/golang-1.15-src_1.15.9-6_amd64.deb  404  Not Found [IP: 146.75.38.132 80]
E: Failed to fetch http://deb.debian.org/debian/pool/main/g/golang-1.15/golang-1.15-go_1.15.9-6_amd64.deb  404  Not Found [IP: 146.75.38.132 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

Exited with code exit status 100

File: pyproject.toml [hunk 1]
```diff
@@ -27,31 +29,50 @@ classifiers = [
...
+files = [
+    "src/",
+    "tests/",
+]
+
+[tool.ruff]
+ignore = [
+    "E722", # Do not use bare `except`
+    "E501", # Line too long (104 > 88 characters)
+#    "E731", # Do not assign a `lambda` expression, use a `def`
```
Comment (y26805) [inline pyproject.toml]: why is this commented out?

**Review by y26805 — APPROVED** [review]
changes look sane

File: pyproject.toml (hunk 1 above)
Comment (ojii) [reply pyproject.toml]: forgot to remove it

**Review by jgaul3 — APPROVED** [review]

**Pull request merged**